testsuite/tools/parsecmm.mly

```diff
@@ -280,7 +280,7 @@ expr:
           Debuginfo.none) }
   | LPAREN ADDRASET expr expr expr RPAREN
       { let open Lambda in
-        Cop(Cstore (Word_val, Assignment),
+        Cop(Cstore (Word_int, Assignment),
```

Comment:
I don't think it's the right fix. I think that either generating a call to `caml_modify` or removing the `addraset` operation completely would be better.
From a quick look at the cmm tests, `arith.cmm` should probably be using `intaset`, `immediates.cmmpp` as well, and `tagged-quicksort.cmm` looks like it is only manipulating integer arrays so it could use `intaset` too.